ui/src/variables/reducers/editor.ts

```diff
@@ -0,0 +1,56 @@
+// Libraries
+import {produce} from 'immer'
+
+// Types
+import {VariableEditorState} from 'src/types'
+import {
+  EditorAction,
+  CLEAR_VARIABLE_EDITOR,
+  CHANGE_VARIABLE_EDITOR_TYPE,
+  UPDATE_VARIABLE_EDITOR_NAME,
+  UPDATE_VARIABLE_EDITOR_QUERY,
+  UPDATE_VARIABLE_EDITOR_MAP,
+  UPDATE_VARIABLE_EDITOR_CONSTANT,
+} from 'src/variables/actions/creators'
+
+export const initialEditorState = (): VariableEditorState => ({
+  name: '',
+  selected: 'query',
+  argsQuery: null,
+  argsMap: null,
+  argsConstant: null,
+})
+
+export const variableEditorReducer = (
+  state: VariableEditorState = initialEditorState(),
+  action: EditorAction
+): VariableEditorState =>
+  produce(state, draftState => {
+    switch (action.type) {
+      case CLEAR_VARIABLE_EDITOR: {
+        return initialEditorState()
```

Comment:
does `draftState` need to be set to `initialEditorState` here? this case is kinda different in form from all the other cases.

Reply:
nope! https://immerjs.github.io/immer/docs/return

Reply:
nope!  https://immerjs.github.io/immer/docs/return